src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Color ='red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+    this.shape = 'triangle';
 
-export class Circle {
+    if (this.c >= this.a + this.b) {
```

Comment:
But what if a or b is bigger than sum of other sides?
Also add check for all sides being larger than 0